In [1]:
import pandas as pd

In [15]:
df = pd.read_csv('../data/credit-data.csv')
df.head()

i#clientid        income        age         loan  c#default
0           1  66155.925095  59.017015  8106.532131          0
1           2  34415.153966  48.117153  6564.745018          0
2           3  57317.170063  63.108049  8020.953296          0
3           4  42709.534201  45.751972  6103.642260          0
4           5  66952.688845  18.584336  8770.099235          1

## Pré-processamento

In [23]:
from sklearn.impute import SimpleImputer
from sklearn.preprocessing import StandardScaler
from sklearn.model_selection import train_test_split

In [16]:
df.loc[df['age'] < 0, 'age'] = df['age'].mean()

In [17]:
previsores = df.iloc[:, 1:4].values
classe = df.iloc[:, 4].values
print(previsores[:2], classe[:2], sep='\n')

[[6.61559251e+04 5.90170151e+01 8.10653213e+03]
 [3.44151540e+04 4.81171531e+01 6.56474502e+03]]
[0 0]


In [19]:
simputer = SimpleImputer()
simputer = simputer.fit(previsores)
previsores = simputer.fit_transform(previsores)

In [20]:
scaler = StandardScaler()
previsores = scaler.fit_transform(previsores)

In [21]:
previsores[:2]

array([[ 1.45393393,  1.36539359,  1.20281942],
       [-0.76217555,  0.54267291,  0.69642695]])

In [27]:
previsores_treinamento, previsores_teste, classe_treinamento, classe_teste  = train_test_split(previsores, classe, test_size= .25, random_state=0)

## SVM

In [26]:
from sklearn.svm import SVC

In [41]:
classificador = SVC(kernel='rbf', C=2, random_state=0)
classificador = classificador.fit(previsores_treinamento, classe_treinamento)
previsoes = classificador.predict(previsores_teste)

/home/dstrb/anaconda3/envs/az_ml/lib/python3.6/site-packages/sklearn/svm/base.py:196: FutureWarning: The default value of gamma will change from 'auto' to 'scale' in version 0.22 to account better for unscaled features. Set gamma explicitly to 'auto' or 'scale' to avoid this warning.
  "avoid this warning.", FutureWarning)


## Métricas

In [42]:
from sklearn.metrics import accuracy_score, confusion_matrix

In [43]:
print(accuracy_score(classe_teste, previsoes), confusion_matrix(classe_teste, previsoes), sep='\n')

0.988
[[434   2]
 [  4  60]]


kernel linear: 0.946 <br>
kernel poly: 0.968 <br>
kernel sigmoid: 0.838 <br>
kernel rbf: 0.98 <br>

c 2 = 0.988<br>